scarb/src/core/package/name.rs

```diff
@@ -88,9 +88,89 @@ impl PackageName {
             }
         }
 
+        if PackageName::is_keyword(&name) {
+            bail!(
+                "the name `{name}` cannot be used as a package name, \
+                names can't use cairo keywords see the full list at \
+                https://docs.cairo-lang.org/language_constructs/keywords.html and \
+                https://github.com/starkware-libs/cairo/blob/dc7f0d5ca76b7409816d99772033abf58975a872/crates/cairo-lang-parser/src/lexer.rs#L184-L209"
+            )
+        }
+
+        if ["test", "scarb"].contains(&name.as_ref()) {
+            bail!("the name `{name}` cannot be used as a package name")
+        }
+
         Ok(Self(name))
     }
 
+    /// Checks if name is a Cairo keyword
+    pub fn is_keyword(name: impl AsRef<str>) -> bool {
```

Comment:
```suggestion
    pub fn is_keyword(name: &str) -> bool {
```

we don't have to use generics here, aren't we?